pkg/nodeconfig/nodeconfig.go

```diff
@@ -172,14 +173,10 @@ func (nc *nodeConfig) Configure() error {
 
 		// Ensure we are annotating the node as soon as the Node object is created, so that we can identify which
 		// controller should be watching it
-		nc.addAdditionalAnnotations()
-		nc.addPubKeyHashAnnotation()
-		node, err := nc.k8sclientset.CoreV1().Nodes().Update(context.TODO(), nc.node, meta.UpdateOptions{})
-		if err != nil {
-			return errors.Wrapf(err, "error updating public key hash and additional annotations on node %s",
-				nc.node.GetName())
+		nc.additionalAnnotations[PubKeyHashAnnotation] = nc.publicKeyHash
+		if err := nc.addAnnotations(nc.additionalAnnotations); err != nil {
```

Comment:
why do we need to pass in nc.additionalAnnotations here? can't we just retrieve it at the call site from the struct, as it is already assigned a value.

Reply:
`addAnnotations` is more of a helper that just handles what is passed in instead of retrieving annotations from `nc` as the function is used multiple times for different annotations.